### PR TITLE
Check for pre-existing details before updating details

### PR DESCRIPTION
**Pull request by SAKET687**
Check for pre-existing details before updating in the API. Please review this and let me know if anything needs to be changed.
Added API for changing passwords for both user and restaurant.

 @arka1997 && @SaiAshish1729

### CODE REVIEW

File: src/controller/restaurant.js [hunk 1]
```diff
@@ -660,9 +660,82 @@ const restaurantProfileUpdate = async (req, h) => {
 	}
 };
 
+// change password with existing password
+const changeRestaurantPassword = async (req, h) => {
+	try {
+		const restaurant = req.rootUser;
+		if (!restaurant) {
```
Comment (SaiAshish1729) [inline src/controller/restaurant.js:667]: Don't need to check this

File: src/controller/restaurant.js [hunk 2]
```diff
@@ -660,9 +660,82 @@ const restaurantProfileUpdate = async (req, h) => {
...
+
+		const existingRestaurant = await prisma.restaurant.findFirst({
+			where: {
+				id: restaurant.id,
+				is_active: true,
+				deleted_at: null,
+			},
+		});
+
+		if (!existingRestaurant) {
```
Comment (SaiAshish1729) [inline src/controller/restaurant.js:685]: Remove this check

File: src/controller/restaurant.js (hunk 1 above)
Comment (SAKET687) [reply src/controller/restaurant.js]: Removed this check.

File: src/controller/restaurant.js (hunk 2 above)
Comment (SAKET687) [reply src/controller/restaurant.js]: Removed this check.

File: src/Routes/UserRoute.js [hunk 3]
```diff
@@ -1,119 +1,252 @@
...
+				},
+			},
+		},
+	},
 
-    },
-]
\ No newline at end of file
+	// profile deletion by user
+	{
```
[the remark this answers is not in the document]
Comment (SAKET687) [reply src/Routes/UserRoute.js:178]: Done

File: src/Routes/UserRoute.js [hunk 4]
```diff
@@ -1,119 +1,252 @@
...
+					payloadType: "json",
+					responseMessages: [],
+				},
+			},
+		},
+	},
+
+	// profile deletion by admin
+	{
+		method: "POST",
```
[the remark this answers is not in the document]
Comment (SAKET687) [reply src/Routes/UserRoute.js]: Done

**Review by SaiAshish1729 — APPROVED** [review]

**Pull request merged**